Audit config section, populate advanced-config section (#167)
filecoin-project/lotus-docs: content/en/storage-providers/setup/prerequisites.md

---
title: "Prerequisites"
description: "This guide describes the necessary prerequisites before configuring a Lotus miner for production."
lead: "This guide describes the necessary prerequisites before configuring a Lotus miner for production.."
draft: false
menu:
    storage-providers:
        parent: "storage-providers-setup"
        identifier: "storage-providers-setup-prerequisites"
aliases:
    - /docs/storage-providers/setup/
weight: 100
toc: true
---

> Mining will only work if you fully cover the [minimal hardware requirements]({{< relref "hardware-requirements" >}}) for the network in which you will mine. As the mining process is very demanding for the machines on several aspects and relies on precise configuration, we strongly recommend Linux systems administration experience before embarking.

Reviewer: ```suggestion
Mining will only work if you fully comply with the [minimal hardware requirements]({{< relref "hardware-requirements" >}}) for the network in which you will mine. The mining process is very resource-intensive and is dependent on precise configuration. We strongly recommend Linux systems administration experience before embarking.
```

lotus wallet new bls
f3...
```

{{< alert icon="callout" >}}
Next make sure to [send some funds]({{< relref "manage-fil" >}}) to the **worker address** so that the miner setup can be completed. The amount you should initialize with varies with gas fees, but 0.1 FIL is generally a safe amount. The sender doesn't have to be any particular address and can be specified using the `from` flag. If `from` is unspecified, the sender will default to the `owner` address, in which case the `onwer` must have the 0.1 FIL. If the `owner` is also unspecified, the wallet's default address is used as the owner and that address must have the 0.1 FIL. 
{{< /alert >}}

For additional information about the different wallets that a miner can use and how to configure them, read the [miner addresses guide]({{< relref "addresses" >}}).

{{< alert icon="tip" >}}
Safely [backup your wallets]({{< relref "manage-fil#exporting-and-importing-addresses" >}})!
{{< /alert >}}

## Optional prerequisites

These prerequisites are optional and can be used on case by case basis. Please make sure to understand the use case before performing these steps.

### Install extra dependencies

> This step is only if you are running an Nvidia GPU and would prefer to use OpenCL instead of CUDA. Please note that Cuda is after than OpenCL for sealing and proving workload. Most Linux distributions contain this package in their package manager:

Reviewer: ```suggestion
This step is only necessary if you are running an Nvidia GPU and would prefer to use OpenCL instead of CUDA. Please note that Cuda is recommended over OpenCL for sealing and proving workload. Most Linux distributions contain this package in their package manager:
```

```shell
sudo apt update -y && sudo apt install -y nvidia-opencl-icd -y
```

### Optional Performance tweaks

> ```sh

Reviewer 2: ```suggestion
```shell
```

# See https://github.com/filecoin-project/bellman
export BELLMAN_CPU_UTILIZATION=0.875
```

The `BELLMAN_CPU_UTILIZATION` is an optional variable to designate a proportion of the multi-exponentiation calculation to be moved to a CPU in parallel to the GPU. This is an effort to keep all the hardware occupied. The interval must be a number between `0` and `1`. The value `0.875` is a good starting point, but you should experiment with it if you want an optimal setting. Different hardware setups wil
l result in different values being optimal. Omitting this environment variable might also be optimal.

### Downloading parameters

To download the parameters do:

> ```sh

Reviewer 2: ```suggestion
```shell
```
